sdk/appconfiguration/app-configuration/src/internal/helpers.ts

```diff
@@ -211,14 +211,19 @@ export function serializeAsConfigurationSettingParam(
     | ConfigurationSettingParam<FeatureFlagValue>
     | ConfigurationSettingParam<SecretReferenceValue>
 ): ConfigurationSettingParam {
-  if (isConfigSettingWithFeatureFlagValue(setting)) {
-    return FeatureFlagHelper.toConfigurationSettingParam(setting);
-  }
-  if (isConfigSettingWithSecretReferenceValue(setting)) {
-    return SecretReferenceHelper.toConfigurationSettingParam(setting);
+  let couldNotEncode = false;
+  try {
+    if (isConfigSettingWithFeatureFlagValue(setting)) {
+      return FeatureFlagHelper.toConfigurationSettingParam(setting);
+    }
+    if (isConfigSettingWithSecretReferenceValue(setting)) {
+      return SecretReferenceHelper.toConfigurationSettingParam(setting);
+    }
+  } catch (error) {
+    couldNotEncode = true;
   }
-  if (isSimpleConfigSetting(setting)) {
-    return setting;
+  if (isSimpleConfigSetting(setting) || couldNotEncode) {
```

Comment:
It feels like if we change up the order of these statements it might simplify things.

For instance, if we check isSimpleConfigSetting(setting) _first_ then the try/catch below can go away and you can just let each function throw instead and not have to do the `couldNotEncode` check.

This would also allow you (in your test) to make sure we're not eating any exceptions that we don't intend to.

Reply:
ok

Reply:
updated as suggested